### DISCUSSION TITLE
Why is PlexTraktSync manually scrobbling to metadata.provider.plex.tv?

### DISCUSSION THREAD

**Discussion by vanstinator** [Taxel/PlexTraktSync#1842] [General]
👋🏻 Hey there! I'm an engineer at Plex, and we noticed that this plugin is making requests to `metadata.provider.plex.tv` to mark media as watched/unwatched. We're preparing to remove those handlers from that service as they're deprecated internally, and wanted to reach out to better understand the use-case you have, and to give you a heads up before the handlers stop functioning. 

As far as I can tell this tool only interacts with Plex Media Server? If that's the case, simply marking the media item as watched/unwatched in PMS will trigger PMS' [View State Sync](https://support.plex.tv/articles/sync-watch-state-and-ratings/) to synchronize the watch state with the cloud, manually sending the action to `metadata.provider.plex.tv` is unnecessary. 

Is there another use-case that requires hitting the cloud endpoints?

**Comment by simonc56** [comment]
PlexTraktSync uses Plex cloud endpoint to sync user watchlist :
- https://python-plexapi.readthedocs.io/en/latest/modules/mixins.html#plexapi.mixins.WatchlistMixin.addToWatchlist

But not for watched/unwatched sync, as far as I know.

About the deprecated `metadata.provider.plex.tv` url, we discussed it here :
- https://github.com/pkkid/python-plexapi/pull/1239

Some parts of the code haven't been updated though :
- https://github.com/search?q=repo%3ATaxel%2FPlexTraktSync+metadata.provider.plex.tv&type=code

@glensc any thought ?

**Reply by glensc** [reply]
@vanstinator what makes you think this app is making such requests? can you provide more details?

@simonc56 those are for creating link to display info about movie/episode. the links are displayed in terminal, user has to actually click them. app is not making those requests itself.

markPlayed/markUnplayed got implemented in python-plexapi:
- https://github.com/pkkid/python-plexapi/pull/1090

but there's no code to use that yet, only your test pr which I doubt many use:
- https://github.com/Taxel/PlexTraktSync/pull/1676

**Reply by vanstinator** [reply]
We see a lot of traffic to our actions endpoints in our monitoring. I see the watchlist actions too. We'll take another pass through our monitoring data and make sure we haven't conflated the python user agent anywhere.

**Reply by glensc** [reply]
Samples of exact urls and http methods?

and what is "a lot of traffic"? at what rate? or percentage of overall traffic of this app?

there's "watch" subcommand that does websocket listen and may query local plex server, I don't directly see how it could make requests to metadata domain.

**Reply by glensc** [reply]
@vanstinator can you show sample of those requests? I checked my logs and it's only making GET requests there.

**Reply by glensc** [reply]
reported dissapeared (and they didn't chose answer). closing
